lib/simulator/SimulatorLightController.cpp

```diff
@@ -10,92 +8,58 @@
-        lights[pole_index][x][y]->setSpecularColour(
-            grid_lights[x][y].r / 255.0 * kLightScale,
-            grid_lights[x][y].g / 255.0 * kLightScale,
-            grid_lights[x][y].b / 255.0 * kLightScale);
-        lights[pole_index][x][y]->setDiffuseColour(
-            grid_lights[x][y].r / 255.0 * kLightScale,
-            grid_lights[x][y].g / 255.0 * kLightScale,
-            grid_lights[x][y].b / 255.0 * kLightScale);
-      }
-    }
-  }
-}
+    leds[pole_number * kLedsPerOutput + 0] = grid_lights[0][0];
+    leds[pole_number * kLedsPerOutput + 1] = grid_lights[0][1];
+    leds[pole_number * kLedsPerOutput + 2] = grid_lights[0][2];
+    leds[pole_number * kLedsPerOutput + 3] = grid_lights[0][3];
 
-Ogre::Light *SimulatorLightController::createLight(
-    Ogre::Vector3 const position) {
-  Ogre::Light *spotLight = scnMgr->createLight();
-  spotLight->setType(Ogre::Light::LT_SPOTLIGHT);
-  spotLight->setDirection(Ogre::Vector3::NEGATIVE_UNIT_Z);
-  // Default lights to off
-  spotLight->setDiffuseColour(0, 0, 0);
-  spotLight->setSpecularColour(0, 0, 0);
+    leds[pole_number * kLedsPerOutput + 4] = grid_lights[1][0];
+    leds[pole_number * kLedsPerOutput + 5] = grid_lights[1][1];
+    leds[pole_number * kLedsPerOutput + 6] = grid_lights[1][2];
+    leds[pole_number * kLedsPerOutput + 7] = grid_lights[1][3];
 
-  Ogre::SceneNode *spotLightNode =
-      scnMgr->getRootSceneNode()->createChildSceneNode();
-  spotLightNode->attachObject(spotLight);
-  spotLightNode->setDirection(0, 0, -1);
-  spotLightNode->setPosition(position);
-  spotLightNode->lookAt(Ogre::Vector3(0, inchesToCoords(30), 0),
-                        Ogre::Node::TransformSpace::TS_WORLD);
+    leds[pole_number * kLedsPerOutput + 8] = grid_lights[2][0];
+    leds[pole_number * kLedsPerOutput + 9] = grid_lights[2][1];
+    leds[pole_number * kLedsPerOutput + 10] = grid_lights[2][2];
+    leds[pole_number * kLedsPerOutput + 11] = grid_lights[2][3];
 
-  spotLight->setSpotlightRange(Ogre::Degree(60), Ogre::Degree(75));
-
-  // Not actually measured, just a wild guess
-  spotLight->setSpotlightFalloff(0.5);
-
-  // Make light get less bright farther away from the light sources.
-  spotLight->setAttenuation(2000, /* constant */ 1.0, /* linear */ 0.0007,
-                            /* quadratic */ 0.0);
-
-  return spotLight;
+    leds[pole_number * kLedsPerOutput + 12] = grid_lights[3][0];
+    leds[pole_number * kLedsPerOutput + 13] = grid_lights[3][1];
+    leds[pole_number * kLedsPerOutput + 14] = grid_lights[3][2];
+    leds[pole_number * kLedsPerOutput + 15] = grid_lights[3][3];
+  }
 }
 
-float SimulatorLightController::inchesToCoords(float inches) {
-  // 64" == 5'4". Model is 192 units tall.
-  return inches * 192.0 / 64.0;
+void SimulatorLightController::SetLedPositions() {
+  for (int pole = 0; pole < Pole::kNumPoles; pole++) {
+    for (int x = 0; x < gridWidth; x++) {
+      for (int y = 0; y < gridHeight; y++) {
```

Comment:
I would prefer to reverse these two loops, so that we consistently visit the y coordinate, then the x. This would mean we travel forward through memory rather than jumping around.

Reply:
Done.